esphome/components/ble_client/output/__init__.py

```diff
@@ -61,7 +63,8 @@ def to_code(config):
             config[CONF_CHARACTERISTIC_UUID]
         )
         cg.add(var.set_char_uuid128(uuid128))
-
+    if CONF_REQUIRE_RESPONSE in config:
+        cg.add(var.set_require_response(config[CONF_REQUIRE_RESPONSE]))
```

Comment:
Default is `False` so conditional not needed
```suggestion
    cg.add(var.set_require_response(config[CONF_REQUIRE_RESPONSE]))
```